tests/integration/cmd_link_unlink_test.go

```diff
@@ -9,138 +13,201 @@ import (
+			When("the link is pushed", func() {
+				JustBeforeEach(func() {
+					helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
+				})
+				It("should ensure that the proper envFrom entry was created", func() {
+					envFromOutput := runner.GetEnvFromEntry(frontendComp, "app", commonVar.Project)
+					Expect(envFromOutput).To(ContainSubstring(backendComp))
+					helper.HttpWaitFor(frontendURL, "Hello world from node.js!", 20, 1)
+				})
+				It("should find the link and environment variables in odo describe", func() {
+					checkDescribe(frontendContext, backendComp, true, false)
+				})
+				It("should find the linked environment variable", func() {
+					stdOut := helper.Cmd("odo", "exec", "--context", frontendContext, "--", "sh", "-c", "echo $SERVICE_BACKEND_IP").ShouldPass().Out()
+					Expect(stdOut).To(Not(BeEmpty()))
+				})
+				It("should not allow re-linking", func() {
+					outputErr := helper.Cmd("odo", "link", backendComp, "--context", frontendContext).ShouldFail().Err()
+					Expect(outputErr).To(ContainSubstring("already linked"))
+				})
+
+				It("should successfully delete component after linked component is deleted", func() {
+					// Testing: https://github.com/openshift/odo/issues/2355
+					helper.Cmd("odo", "delete", "-f", "--context", backendContext).ShouldPass()
+					helper.Cmd("odo", "delete", "-f", "--context", frontendContext).ShouldPass()
+				})
+
+				When("unlinking the two components", func() {
+					JustBeforeEach(func() {
+						helper.Cmd("odo", "unlink", backendComp, "--context", frontendContext).ShouldPass()
+					})
+					It("should find the link in odo describe", func() {
+						checkDescribe(frontendContext, backendComp, true, false)
+					})
+					It("should not allow unlinking again", func() {
+						stdOut := helper.Cmd("odo", "unlink", backendComp, "--context", frontendContext).ShouldFail().Err()
+						Expect(stdOut).To(ContainSubstring(fmt.Sprintf("failed to unlink the component %q since no link was found in the configuration referring this component", backendComp)))
+					})
+
+					When("odo push is executed", func() {
+						JustBeforeEach(func() {
+							helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
+						})
+						It("should no longer find the link in odo describe", func() {
+							stdOut := helper.Cmd("odo", "describe", "--context", frontendContext).ShouldPass().Out()
+							Expect(stdOut).ToNot(ContainSubstring("Linked Services"))
+							Expect(stdOut).ToNot(ContainSubstring(backendComp))
+						})
+					})
+				})
+			})
+			It("should unlinking a non-pushed link successfully", func() {
+				helper.Cmd("odo", "unlink", backendComp, "--context", frontendContext).ShouldPass()
+			})
+		})
+		When("a link is created between the two components with --bind-as-files", func() {
+			JustBeforeEach(func() {
+				helper.Cmd("odo", "link", backendComp, "--bind-as-files", "--context", frontendContext).ShouldPass()
 			})
```

Comment:
Lets get the [scenario](https://github.com/openshift/odo/pull/4891/files#diff-74fad2691ab8db201db69243e05dfc16337080032e40d791d3da4a6753d69d43R208-R209) just after BeforeEach part. Only reason is better visibility, as it is aligned to this particular when only.